### ISSUE TITLE
Pre-request GET with JSON body results in “error”:“No query parameter provided”

### ISSUE THREAD

**Issue by mickmcbride** [a85/POSTMan-Chrome-Extension#10116]
I'm trying to do a GET in Pre-request that contains a body. It fails with a 400 error, `{"error": "No query parameter provided"}`. If I do the request normally with the { "email": email } set in the Body and raw JSON set it returns with a 200 and data. I'm not sure why it is having an issue when it is sent in the Pre-request. 

Other GETs to the same API that don't have the Body work correctly in the Pre-request. I'm able to use curl, request, Axios and all return successfully. 
        
Here is the Pre-request that I am trying:

```javascript
pm.sendRequest({
    method: 'GET',
    url: pm.collectionVariables.get("base_url") + '/api/v1/users/with_params',
    header: {
        'AuthToken': pm.collectionVariables.get("token"),
        'Content-Type': 'application/json'
    },
    body: {
        mode: 'raw',
        raw: JSON.stringify({
            "email": "email@email.com"
        })
    }
}, function (err, response) {
    console.log(response);
});
```

**Comment by postman-staff** [comment]
Hi,

Can you please if the endpoint needs the query parameter?

via @ash007-postman | Postman Support

**Comment by mickmcbride** [comment]
The endpoint does not take any query parameters or path variables.

**Comment by postman-staff** [comment]
Hi,

We are unable to replicate the issue. Looks like the error `"No query parameter"` is coming from your API. Can you share the screenshot of the error?

Also would like to inform you that for How-To questions, our community is the best place: [Postman Community](https://community.postman.com) - it is a vibrant community frequented by industry experts and our engineers alike for queries that are API-specific or any implementation questions.

via @ash007-postman | Postman Support

**Comment by mickmcbride** [comment]
Hello,

The error is being returned from our API because it is not getting the correct data with the request. 

When I send it in a normal request using Postman, curl, etc in Datadog it shows:
`I, [2021-07-16T19:49:19.757553 #65]  INFO -- : []   Parameters: {"email"=>"case-2-s_damiani@migration-example-icloud.com", "user"=>{"email"=>"case-2-s_damiani@migration-example-icloud.com"}}`

When I send it in a pre-request in Postman in Datadog it shows:
`I, [2021-07-16T19:41:55.367068 #65]  INFO -- : []   Parameters: {"user"=>{}}`

This makes me assume that Postman is not understanding pre-request GETs that contain a JSON body. 

I have posted the question to the Postman Community with no response and as I assume that it probably isn't a configuration issue I posted it here. 

Do you have a working example of a GET with a JSON body? Possibly in Postman echo?

Thank you

**Comment by postman-staff** [comment]
Thanks for reaching back.

I reviewed the Pre-request script you use to send the GET request, and it doesn’t seem that there’s anything wrong with it. I have tested it on my end and it looks like it’s working successfully, and the body seems to be carrying a JSON payload same output can be seen if you send the request in a JSON format and providing a body to it. Can you please try opening Postman’s console, and see whether the Request Body is being passed? if possible can you please attach screenshots of a successful request once you try to reach the service in Postman, that way we can try to spot what the difference between the two cases is? Just make sure to include both the request and the response in the screenshot.

 ![](https://postman.zendesk.com/attachments/token/dxIdHyquzsoCWlE6x7gfH8bqZ/?name=inline-1757692601.png)​
via @ash007-postman | Postman Support

**Comment by mickmcbride** [comment]
I just tested using the postman-echo endpoint as you demonstrated and it is not sending the "args" in the pre-request request either, the same as I am having with this issue. Please see the attached screenshot of the difference between a normal request and pre-request request. If you still need screenshots from my API I can attach them as well. 

![Screen Shot 2021-07-19 at 8 52 05 AM](https://user-images.githubusercontent.com/57964844/126183611-3742fdb5-4584-41f5-a37c-edf7bd7b80d9.png)

**Comment by postman-staff** [comment]
Hi,

Thanks for sharing the screenshots. We have tested this again, also checked it with the respective team and it looks like as per the workflow you can't send a body with the following methods `get, copy, head, purge, unlock` using with the pm.sendRequest API which means we handle it via the configuration in the application but that is not possible with the pm.sendRequest API.

I hope this clarifies.

via @ash007-postman | Postman Support

**Comment by mickmcbride** [comment]
Hello,

A GET request with a JSON body is a valid request. The pm.sendRequest API not supporting all valid requests is an issue.

**Comment by codenirvana** [comment]
@mickmcbride we've pushed a fix for this in the latest Postman app (v9.9.0). Can you update your app and confirm if that fixes your issue?

**Comment by mickmcbride** [comment]
@codenirvana I tested with 9.8.3, not working, upgraded to 9.9.3 and it is working correctly. 

Thank you,
Mick
